src/ahp/prover.rs

```diff
@@ -356,14 +360,16 @@ impl<F: PrimeField> AHPForR1CS<F> {
 
         let z_a_poly_time = start_timer!(|| "Computing z_A polynomial");
         let z_a = state.z_a.clone().unwrap();
-        let z_a_poly = &EvaluationsOnDomain::from_vec_and_domain(z_a, domain_h).interpolate()
-            + &(&DensePolynomial::from_coefficients_slice(&[F::rand(rng)]) * &v_H);
+        let r_a = F::rand(rng);
```

Comment:
Can there be a comment here explaining the operation these 3 lines are doing? (Ditto for z_b)

I believe its "go from z_a evaluations to z_a_poly as `interpolate(z_a_poly | H) * (Z_I)) + rand_poly`

Reply:
It's computed as `interpolate(z_a_evaluations_over_H) + v_H * r_a`